### ISSUE TITLE
Support for ResourceDiscoveryMode in `azurerm_policy_remediation`

### ISSUE THREAD

**Issue by ArcturusZhang** [terraform-providers/terraform-provider-azurerm#6658]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

Support for ResourceDiscoveryMode option in `azurerm_policy_remediation`. Reference: https://github.com/Azure/azure-sdk-for-go/blob/master/services/policyinsights/mgmt/2019-10-01/policyinsights/models.go#L2056

### New or Affected Resource(s)

* azurerm_policy_remediation

### Potential Terraform Configuration

```hcl
# Copy-paste your Terraform configurations here - for large Terraform configs,
# please use a service like Dropbox and share a link to the ZIP file. For
# security, you can also encrypt the files using our GPG public key.
```

### References

* #0000

**Comment by t3mi** [comment]
Should be closed as [#9210](https://github.com/terraform-providers/terraform-provider-azurerm/pull/9210) landed.

**Comment by tombuildsstuff** [comment]
Fixed via #9210 - thanks for the heads up @t3mi
